source/language/gates.rst

```diff
@@ -378,7 +294,87 @@ repetitions of ``inv @ U`` for :math:`k < 0`.
+
+Global phase gate ``gphase``
+++++++++++++++++++++++++++++
+
+From a physical perspective, the unitaries :math:`e^{i\gamma}V` and :math:`V` are equivalent although they differ by a global
+phase :math:`e^{i\gamma}`. When we add a control to these gates, however, the global phase becomes a relative phase
+that is applied when the control qubit is one. A built-in global phase gate
+allows the inclusion of arbitrary global phases on circuits. The instruction ``gphase(γ);`` accumulates a global phase
+of :math:`e^{i\gamma}`.
+
+Just as every n-qubit gate can be thought of as generating a tensor product with the suitable
+identity matrix to cover all other qubits in the gate, subroutine, or global scope containing the
+instruction, similarly ``gphase`` behaves as a 0-qubit gate and when applied in a context with
+`m` qubits in scope, behaves as applying the unitary
+
+.. math::
+   \operatorname{gphase}(\gamma) := e^{i\gamma} I_m,
+
+where :math:`I_m` denotes the identity matrix with size :math:`2^m`
+
+For example
+
+.. code-block::
+
+   gate X q {
+      U(π, 0, π) q;
+      gphase -π/2;
+   }
+
+   gate CX c, t {
+      ctrl @ X c, t;
+   }
+
+defines ``CX`` as the standard CNOT gate.
+
+Relation of the built-in gates to hardware-native gates
+----------------------------
+
+For *non-parameterized gates*, the choice of ``U`` and ``gphase`` as the built-in gates, along with one
+two-qubit entangling gate CNOT as defined gives a universal gate set that can represent general n-qubit
+unitaries with an :math:`O(2^n)` size description :cite:`barenco95`. This basis is not an enforced compilation
+target but a mechanism to define other gates. For many gates of
+practical interest, there is a circuit representation with a polynomial
+number of one- and two-qubit gates, giving a more compact representation
+than requiring the programmer to express the full :math:`2^n \times 2^n`
+matrix. However, a general :math:`n`-qubit gate can be defined using an
+exponential number of these gates. Thus there is no particular privilege incurred by hardware implementations
+that natively support the built-in gates.
+
+For *parameterized gates*, the choice of built-in gates *does* constrain which hardware-native gates are well-
+supported, because conversion between parameterized basis sets in general can be involved, requiring careful
+selection of branch cuts and other logic that would not likely be feasible to specify as compact mathematical
+expressions, nor to evaluate at runtime for cases where the parameters depend on quantum measurements.
+
+For many current platforms the qubits are defined relative to a
+rotating frame and the rotating wave approximation (RWA) holds. This is the domain covered by the OpenPulse
+specification. For this case, the only supported form of run-time parameterization
+will likely be via a ``rz(ϕ)`` implemented by specialized frame-tracking hardware.
+This gate is covered by the built-in ``U`` as a special case ``U(0, ϕ, θ)``
```

Comment:
Do you mean to have zeros for the first two arguments of `U`?

Reply:
I didn't but agree that would be clearer